### PR TITLE
Clear the lastLimboFreeSnapshot version on existence filter mismatch

### PR DESCRIPTION
**Pull request by schmidt-sebastian**
This fixes a problem where the SDK receives an existence filter mismatch and clears the mapping of documents to targets, but does not receive another snapshot that also clears the snapshot version (i.e. the device went in the background and did was not able to actually process a re-run of the query).

The change that this PR makes is that we always reset the lastLimboFreeSnapshot version to 0 if we get a target mismatch.

Fixes https://github.com/firebase/firebase-android-sdk/issues/3249

### CODE REVIEW

**Review by bswhite1 — COMMENTED** [review]
Should the test case in existence_spec_test.json, or somewhere else, check that the targetData.lastLimboFreeSnapshotVersion is reset to none?

Other than that, the code itself is good. I verified via several test runs and never saw the error.

**Comment by schmidt-sebastian** [comment]
@bswhite1 Thanks for verifying. I also added another test case.

**Review by bswhite1 — APPROVED** [review]
Looks good.

What is the easiest way to determine when this is part of a release?
Do you ever use git Tags, or release branches?

**Comment by schmidt-sebastian** [comment]
Please take a look at our release notes to determine what is contained in a release. This PR should be included in the next cut.

We use a private GitHub repository to manage our releases. The tags are not visible here.

File: firebase-firestore/CHANGELOG.md [hunk 1]
```diff
@@ -7,6 +7,8 @@ by opting into a release at
   updates that have not yet been synced with the backend.
 - [changed] Improved performance for queries against collections that contain
   subcollections.
+- [fixed] Fixed an issue that may have caused the SDK to raise incomplete
```
Comment (dconeybe) [inline firebase-firestore/CHANGELOG.md:10]: "may have caused" sounds to me like you are unsure that this is the root cause; however, it seems that the root cause is definitively known. Would it make sense to re-word this to imply that the buggy behavior is intermittent?

e.g. Fixed an issue that causes the SDK to sometimes raise incomplete snapshots...

**Review by dconeybe — CHANGES_REQUESTED** [review]

File: firebase-firestore/CHANGELOG.md (hunk 1 above)
Comment (schmidt-sebastian) [reply firebase-firestore/CHANGELOG.md:10]: Good idea. Will update

**Review by dconeybe — APPROVED** [review]

**Pull request merged**